pkg/kn/commands/parsing_helper_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright © 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"reflect"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+func TestMapFromArray(t *testing.T) {
```

Comment:
What about being defensive for user errors, e.g., leaving extra “,” or “=“ at the end as well as other similar cases. At a minimum should error out or warn.

Reply:
A delimiter at the end (e.g. `--label empty=`) is valid for setting an empty string as the value. An extra delimiter is considered a regular character and included in the value (e.g. `--label key=value=` leads to `{"key": "value="}`. I added a few more test cases to showcase this behavior. Is that what you meant or are there other cases you were thinking of that should error/warn?